src/open_inwoner/userfeed/tests/test_external_tasks.py

```diff
@@ -0,0 +1,115 @@
+from django.contrib.auth import user_logged_in
+from django.test import RequestFactory, TestCase
+
+import requests_mock
+from zgw_consumers.test.factories import ServiceFactory
+
+from open_inwoner.accounts.models import User
+from open_inwoner.accounts.tests.factories import DigidUserFactory
+from open_inwoner.openzaak.models import OpenZaakConfig
+from open_inwoner.openzaak.tests.mocks import ESuiteData
+from open_inwoner.openzaak.tests.shared import FORMS_ROOT
+from open_inwoner.userfeed.choices import FeedItemType
+from open_inwoner.userfeed.models import FeedItemData
+from open_inwoner.userfeed.tests.factories import FeedItemDataFactory
+
+
+class UserFeedExternalTasksTestCase(TestCase):
```

Comment:
This case is a good example of my earlier comment about splitting the trigger/decision with the actual functionality. 

Here we have a mix of signal/login stuff, and actual feed update things. Eg: to test the feed update we need to simulate a login.. 

So I'd say do a `test_login_triggers_update()`, with a mocked `update_user_task_feed(user)` that is checked with just an `assert_called_once_with()`. Then we could easily refactor to a `test_render_triggers_update()` if we decide to move it to the plugin render, without having to mess with the feed update testing part that is now intertwined.

Then add one or more `test_update_user_task_feed()` tests to test the update function, expiration etc etc without caring where it was used or mixing in signal or plugin render code.

Reply:
I split it up and changed the login test to a CMS plugin render test. I did also add a check for the display of the FeedItem in the UserFeed.